src/measurement_graph.rs

```diff
@@ -0,0 +1,339 @@
+use crate::{
+    error::MqaError,
+    helpers::{
+        execute_query, named_or_blank_quad_object, named_or_blank_quad_subject, parse_graphs,
+    },
+    measurement_value::MeasurementValue,
+    score::{DimensionScore, MetricScore, Score},
+    vocab::{dcat, dcat_mqa, dqv, rdf_syntax},
+};
+use oxigraph::{
+    io::GraphFormat,
+    model::{
+        vocab::xsd, GraphNameRef, Literal, NamedNode, NamedNodeRef, NamedOrBlankNode,
+        NamedOrBlankNodeRef, Quad, Term,
+    },
+};
+use rand::Rng;
+use regex::Regex;
+use std::{collections::HashMap, io::Cursor};
+
+pub struct MeasurementGraph(oxigraph::store::Store);
+
+impl MeasurementGraph {
+    /// Loads graph from string.
+    pub fn parse<G: ToString>(graph: G) -> Result<Self, MqaError> {
+        let graph = Self::name_blank_nodes(graph.to_string())?;
+        parse_graphs(vec![graph]).map(|store| Self(store))
+    }
+
+    /// Replaces all blank nodes with named nodes.
+    /// Enables SPARQL query with (previously) blank nodes as identifiers.
+    fn name_blank_nodes(graph: String) -> Result<String, MqaError> {
+        let replaced = Regex::new(r"_:(?P<id>[0-9a-f]+) ")
+            .map(|re| re.replace_all(&graph, "<http://blank.node#${id}> "))?;
+        Ok(replaced.to_string())
+    }
+
+    // Undoes replacement of all blank nodes with named nodes.
+    fn undo_name_blank_nodes(graph: String) -> Result<String, MqaError> {
+        let replaced = Regex::new(r"<http://blank.node#(?P<id>[0-9a-f]+)> ")
+            .map(|re| re.replace_all(&graph, "_:${id} "))?;
+        Ok(replaced.to_string())
+    }
+
+    /// Retrieves all named or blank dataset nodes.
+    pub fn dataset(&self) -> Result<NamedOrBlankNode, MqaError> {
+        self.0
+            .quads_for_pattern(None, Some(dcat::DISTRIBUTION.into()), None, None)
```

Comment:
bruk rdf:type og dcat:Dataset isteden for distribution property

Reply:
https://github.com/Informasjonsforvaltning/fdk-mqa-scoring-service/pull/2/commits/4688457693404c5448b20a97dbcfcf5f0cc2ac97